rfcs/0166-nix-formatting.md

```diff
@@ -816,12 +878,10 @@ x
 
 ### with
 
-- If the body is `{ ... }`, `[ ... ]`, `( ... )` or `'' ... ''`, then the body must start on that same line too.
-  - Otherwise, the body of `with attrs;` must start on a new line without any additional indentation.
-- If the body is `{ ... }`, `[ ... ]`, `( ... )` or `'' ... ''`
-  and `with attrs; …` is parenthesised or to the right side of a binding (e.g. `(with; [ …` or `foo = with; [ …`),
-  then `with` must start on the same line.
-  - Otherwise it must start on a new line.
+- `with` expressions of absorbable terms are handled like absorbable terms themselves.
```

Comment:
And this contradicts the example: If `{ ... }` is absorbable, then `with bar; { ... }` is also absorbable according to this sentence, which means it should be this:

```nix
  foo = with foo; with bar; [
    # multiline
    baz
  ];
```

Reply:
> It was pretty clear from the last meeting that it's not just terms that should be treated as absorbable.

Within the context of `with` statements, absorbing anything else does not make much sense.

Within the context of general expressions, the property "absorbable" does not give much value to define, as it is heavily context dependent. (See below)

> And this contradicts the example

No it is not, since the body of `with foo;` is not an absorbable term. The relevant question at hand here is not "is `with bar;` absorbable?", but instead "will `with bar;` be absorbed?". Because the first question would only look at the expression itself and its content. But the second question actually correctly represents the fact that the actual behavior is also dependent on the outer context.

If you want, you can see it as "being absorbable is a necessary condition for absorbtion", however this usually is such a weak condition for anything other than absorbable terms that it does not help that much to formally define "absorbable expression".